packages/cohort/src/tx_batch_executor.rs

```diff
@@ -0,0 +1,120 @@
+use std::str::FromStr;
+
+use crate::bank_api::{AccountUpdate, Transfer};
+use crate::replicator::core::StatemapItem;
+use crate::snapshot_api::SnapshotApi;
+use crate::state::model::{AccountUpdateRequest, BusinessActionType, TransferRequest};
+use crate::state::postgres::database::{Action, Database};
+use deadpool_postgres::GenericClient;
+use futures::future::BoxFuture;
+
+pub struct BatchExecutor {}
+
+impl BatchExecutor {
+    pub async fn execute(db: &Database, batch: Vec<StatemapItem>, snapshot: Option<u64>) -> Result<u64, String> {
```

Comment:
This method ended up to be very complicated. I re-wrote applying early return technique, so that it sieasier to read when there are less nested if statements.

https://github.com/kindredgroup/talos/pull/42/files#diff-5dcb96a8a0c6dfed6b8621c76b47311bcd293f8625856bbe952b2510a9212747R14